src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -114,20 +97,21 @@ private ExpressionSyntax ParseInterpolatedStringToken()
                     var interpolation = interpolations[i];
 
                     // Add a token for text preceding the interpolation
-                    var text = Substring(originalText, (i == 0) ? (openQuoteIndex + 1) : (interpolations[i - 1].CloseBracePosition + 1), interpolation.OpenBracePosition - 1);
+                    var text = originalText[new Range(
+                        i == 0 ? openQuoteIndex + 1 : interpolations[i - 1].CloseBraceRange.End,
+                        interpolation.OpenBraceRange.Start)];
```

Comment:
range math is much saner.  you can just use .Start and .End easily and have things just make sense.